### PR TITLE
Gzip database and exclude

### PR DESCRIPTION
**Pull request by Mte90**
Fix:  #2340 and #2342

Require some testing specifically of the import part.

## Checks

* [x] I've updated the changelog.
* [x] I've tested this PR
* [x] This PR is for the `develop` branch not the `stable` branch.
* [x] This PR is complete and ready for review.

### CODE REVIEW

File: config/homebin/db_backup [hunk 1]
```diff
@@ -36,12 +49,21 @@ do
...
 count=0
 for db in "${databases[@]}"
 do
-    OUTPUT=$(printf "<info>   - %2s/%s Backing up </info><b>%-23s</b><info> to </info><b>'database/backups/%s.sql'</b>" "${count}" "${#databases[@]}" "'${db}'" "${db}")
+    OUTPUT=$(printf "<info>   - %2s/%s Backing up </info><b>%-23s</b><info> to </info><b>'database/backups/%s${ext}'</b>" "${count}" "${#databases[@]}" "'${db}'" "${db}")
     vvv_output "${OUTPUT}"
-    mysqldump -uroot -proot "${db}" > "/srv/database/backups/${db}.sql";
+    if [[ "${gzip}" == "True" ]]; then
+      mysqldump -uroot -proot "${db}" | gzip > "/srv/database/backups/${db}.sql.gz"
+      rm "/srv/database/backups/${db}.sql"
```
Comment (tomjn) [inline config/homebin/db_backup:63]: I don't think we should remove the `sql` files, that would be destructive

File: database/sql/import-sql.sh [hunk 2]
```diff
@@ -41,13 +42,14 @@ cd /srv/database/backups/
...
-sql_count=$(ls -1 ./*.sql 2>/dev/null | wc -l)
+sql_count=$(ls -1 ./*.sql* 2>/dev/null | wc -l)
 if [ "$sql_count" != 0 ]
 then
-	for file in $( ls ./*.sql )
+	for file in $( ls ./*.sql* )
 	do
 		# get rid of the extension
 		pre_dot=${file%%.sql}
+		pre_dot=${file%%.gz}
```
Comment (tomjn) [inline database/sql/import-sql.sh:52]: should these two lines be the other way around?

Comment (Mte90) [reply database/sql/import-sql.sh]: I did a workaround for this

**Comment by Mte90** [comment]
I did some testing and works

File: CHANGELOG.md [hunk 3]
```diff
@@ -21,6 +21,7 @@ permalink: /docs/en-US/changelog/
 * Added improved apt package upgrade routines
 * Provisioners now ask to install only packages that aren't installed
 * General package handling performance improvements
+* New config to exclude database to backup and to gzip the database dump ( #2346 )
```
Comment (tomjn) [inline CHANGELOG.md:24]:
```suggestion
* New config to exclude databases from backup in `config.yml` ( #2346 )
* New config to gzip compress database backups in `config.yml` ( #2346 )
```

**Comment by tomjn** [comment]
This is blocked by a lack of testing, and requires someone who is not the PR author to confirm this works and does not break existing installs

**Comment by Mte90** [comment]
To test:
* Check https://varyingvagrantvagrants.org/docs/en-US/contribute/
Screenshot_20210416_151157

Add settings in your `config.yml` like this. Where instead `demo` there is a website that you don't want to get the backup.
As in the screen with this settings the backup will be compressed and will exclude those websites on generation.

**Comment by tomjn** [comment]
We also need someone to test that this branch doesn't break anything if you make no changes to `config.yml`

**Review by evertiro — APPROVED** [review]
Looks good to me! Good idea!

**Comment by evertiro** [comment]
Errr... Looks good to me once the CHANGELOG is refreshed

**Review by evertiro — APPROVED** [review]
Looks good to me!

**Pull request merged**